### PR TITLE
Tag/Untag Filtered List Bugfix

### LINKED ISSUE #157: Bug: Tag/Untag not working on Filtered Lists
Tag/Untag not working on Filtered Lists, and instead uses the index of the entire list, resulting in unintended behaviour.

### PR DESCRIPTION
**Pull request by krusagiz**
Fixes  #157

### CODE REVIEW

File: src/main/java/thrift/logic/commands/TagCommand.java [hunk 1]
```diff
@@ -85,8 +85,9 @@ public CommandResult execute(Model model) throws CommandException {
                 ? ""
                 : String.format(MESSAGE_TAG_EXISTED, existedTags.toString());
 
-        actualIndex = model.getIndexInFullTransactionList(transactionToTag).get();
-        model.setTransactionWithIndex(index, updatedTransaction);
+        actualIndex = model.getIndexInFullTransactionList(transactionToTag).orElse(index);
+        model.setTransactionWithIndex(actualIndex, updatedTransaction);
+        model.updateFilteredTransactionList(Model.PREDICATE_SHOW_ALL_TRANSACTIONS);
```
Comment (limerencee) [inline src/main/java/thrift/logic/commands/TagCommand.java]: This line is not necessary as we will preserve the current filtered list last updated by the user

File: src/main/java/thrift/logic/commands/UntagCommand.java [hunk 2]
```diff
@@ -85,8 +85,9 @@ public CommandResult execute(Model model) throws CommandException {
                 ? ""
                 : String.format(MESSAGE_TAG_NOT_EXISTED, nonexistentTags.toString());
 
-        actualIndex = model.getIndexInFullTransactionList(transactionToTag).get();
-        model.setTransactionWithIndex(index, updatedTransaction);
+        actualIndex = model.getIndexInFullTransactionList(transactionToTag).orElse(index);
+        model.setTransactionWithIndex(actualIndex, updatedTransaction);
+
         model.updateFilteredTransactionList(Model.PREDICATE_SHOW_ALL_TRANSACTIONS);
```
Comment (limerencee) [inline src/main/java/thrift/logic/commands/UntagCommand.java]: Remove this line as well!

**Review by limerencee — CHANGES_REQUESTED** [review]
Should be able to merge after minor changes.

File: src/main/java/thrift/logic/commands/UntagCommand.java (hunk 2 above)
Comment (krusagiz) [reply src/main/java/thrift/logic/commands/UntagCommand.java]: Removed the requested line.

File: src/main/java/thrift/logic/commands/TagCommand.java (hunk 1 above)
Comment (krusagiz) [reply src/main/java/thrift/logic/commands/TagCommand.java]: Removed the requested line.

**Comment by limerencee** [comment]
LGTM, thank you for the changes and bug fix!

**Review by limerencee — APPROVED** [review]

**Pull request merged**